File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.go
func numsSameConsecDiff(n int, k int) []int {
    

        cur := []int{1, 2, 3, 4, 5, 6, 7, 8, 9}
        for i := 0; i < n - 1; i++ {
            sd := []int{}
            for _, v := range cur {
                n := v % 10
                if n + k < 10 { sd = append(sd, v * 10 + n + k) }
                if n >= k && k > 0 { sd = append(sd, v * 10 + n - k) }
            }
            cur = sd
        }
        return cur

}

File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.py
class Solution:
    def numsSameConsecDiff(self, n: int, k: int) -> List[int]:
        
        
        def backtrack(ix, sd): 
            
            if ix ==n:
                ans.add(sd)
                return 
            
            if ix==0: 
                for i in range(1,10):
                    
                    backtrack(ix+1 , sd+str(i))
            else: 
                t1 = int(sd[-1])+k
                if  t1 <10:
                    backtrack(ix+1, sd+str(t1))
                t2 = int(sd[-1])-k 
                if t2 >=0:
                    backtrack(ix+1 , sd+str(t2))
        
        
         
        ans = set()
        backtrack(0 ,"")
        return ans 